packages/next/server/on-demand-entry-handler.js

```diff
@@ -232,6 +232,10 @@ export default function onDemandEntryHandler (devMiddleware, multiCompiler, {
       const name = join('static', buildId, 'pages', bundleFile)
       const absolutePagePath = pagePath.startsWith('next/dist/pages') ? require.resolve(pagePath) : join(pagesDir, pagePath)
 
+      if (pagePath.indexOf('index.amp') > -1) {
+        page = pagePath.replace('.js', '')
```

Comment:
This should probably be consolidated, considering we already replace via ``new RegExp(`\\.+(?:${pageExtensions.join('|')})$`)`` a few lines above.

Reply:
Yeah, it should be more consolidated now